[Swift 6.0.3] Empty error type on "missing expected error" cases

## Description of the issue

If you try to run `swift test` for this project against a Swift 6.0.3 toolchain, you will get a few warnings, and then weird test errors where an empty type is returned.

There are currently a bunch of small syntax-related issues that break testing out-of-box on 6.0.3 (at least as of commit 43fb175). However, if you remove `Tests/RegoTests/CapabilityEvaluatorTests.swift`, and clean up the 2-3x unexpected comma issues in `BuiltinTests` (some of those are my fault from when I was working from a 6.2 toolchain, and should be corrected soon), you will get a few warnings, of the form:
```
[0/1] Planning build
[1/1] Compiling plugin GenerateDoccReference
[2/2] Compiling plugin GenerateManual
Building for debugging...
[2/27] Write sources
[3/27] Write swift-version-24593BA9C3E375BF.txt
[5/36] Compiling RegoTests AggregatesTests.swift
/workspace/Tests/RegoTests/BuiltinTests/BuiltinTests.swift:58:17: warning: constant 'error' inferred to have type '()', which may be unexpected
 56 |             #expect(successEquals(result, tc.expected))
 57 |         case .failure(let expectedError):
 58 |             let error = try #require(throws: (any Error).self, "Expect an error to be thrown") {
    |                 |- warning: constant 'error' inferred to have type '()', which may be unexpected
    |                 `- note: add an explicit type annotation to silence this warning
 59 |                 try result.get()
 60 |             }
```
and a slew of error messages later on of the general form:
```
... at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
```

## How to repro

 1. `rm Tests/RegoTests/CapabilityEvaluatorTests.swift` (Too many syntax errors in 6.0.3, so I just drop it for the purposes of the repro)
 1. Manually fix the linter errors reported in `Tests/RegoTests/BuiltinTests/BuiltinTests.swift`.
    - To see the Swift 6.0.3 linter errors, try this Docker invocation:
      ```shell
      docker run --rm -v "$PWD:/workspace" -w /workspace swift:6.0.3 swift format lint --strict --parallel --recursive .
      ```
 1. Run the RegoTests suite under Swift 6.0.3 with this Docker invocation:
    ```shell
    docker run --rm -v "$PWD:/workspace" -w /workspace swift:6.0.3 swift test --filter RegoTests
    ```

Those steps should result in a trace that looks like this:

<details>
<summary><b><code>swift test</code> results for Swift 6.0.3</b></summary>

```
% docker run --rm -v "$PWD:/workspace" -w /workspace swift:6.0.3 swift test --filter RegoTests
warning: 'workspace': found 1 file(s) which are unhandled; explicitly declare them as resources or exclude from the target
    /workspace/Tests/RegoTests/CapabilityEvaluatorTests.swift.backup
[0/1] Planning build
[1/1] Compiling plugin GenerateDoccReference
[2/2] Compiling plugin GenerateManual
Building for debugging...
[2/6] Write swift-version-24593BA9C3E375BF.txt
Build complete! (3.28s)
Test Suite 'Selected tests' started at 2026-01-22 21:14:23.872
Test Suite 'Selected tests' passed at 2026-01-22 21:14:23.872
	 Executed 0 tests, with 0 failures (0 unexpected) in 0.0 (0.0) seconds
◇ Test run started.
↳ Testing Library Version: 6.0.3 (18c42c19cac3faf)
◇ Suite "BuiltinRegistry" started.
◇ Suite "BuiltinTests" started.
◇ Suite "ResultSetTests" started.
◇ Suite "IREvaluatorTests" started.
◇ Suite "BundleDecodingTests" started.
◇ Suite "BundleDirectoryLoaderTests" started.
◇ Suite "NSNumber.toNumberRegoValue Extension" started.
◇ Suite BundleLoaderTests started.
◇ Suite StoreTests started.
◇ Suite IRStatementTests started.
◇ Suite PtrTests started.
◇ Suite RelativePathTests started.
◇ Suite "BuiltinsCacheTests" started.
◇ Suite PathTests started.
◇ Suite "BuiltinTests - Rand" started.
◇ Suite "BuiltinTests - Array" started.
◇ Suite "BuiltinTests - Numbers" started.
◇ Suite "BuiltinTests - Crypto" started.
◇ Suite "BuiltinTests - Types" started.
◇ Suite "BuiltinTests - Strings" started.
◇ Suite "BuiltinTests - Time" started.
◇ Suite "BuiltinTests - Objects" started.
◇ Suite "BuiltinTests - Arithmetic" started.
◇ Suite "BuiltinTests - Encoding" started.
◇ Suite "BuiltinTests - Bits" started.
◇ Suite "BuiltinTests - Collections" started.
◇ Suite "BuiltinTests - Casts" started.
◇ Suite "BuiltinTests - Comparison" started.
◇ Suite "BuiltinTests - UUID" started.
◇ Suite "CustomBuiltinTests" started.
◇ Suite "BuiltinTests - Sets" started.
◇ Suite "BuiltinTests - Aggregates" started.
◇ Test evalErrorDoesntPropagateWithoutStrict() started.
◇ Test haltPropagatesOnInvokeNonStrict() started.
◇ Test undefinedAlwaysPropagates() started.
◇ Test haltPropagatesOnInvoke() started.
◇ Test argErrorsDontPropagateWithoutStrict() started.
◇ Test testResultSet(tc:) started.
◇ Test testValidDecodeManifests(tc:) started.
◇ Test testInvalidEvaluations(tc:) started.
◇ Test testValidEvaluations(tc:) started.
◇ Test testInvalidDecodeManifests(tc:) started.
◇ Test testDirectoryLoader(tc:) started.
◇ Test testLoadingBundleFromDirectory(tc:) started.
◇ Test returnsUIntForBoolean() started.
◇ Suite "BuiltinTests - Units" started.
◇ Test testBundleConsistency(tc:) started.
◇ Test returnsIntForNegativeWholeNumbers() started.
◇ Test returnsSelfForFractionalNumbers() started.
◇ Test testStoreReads(tc:) started.
◇ Test testStoreReadsFailures(tc:) started.
◇ Test testStatementEvaluation(tc:) started.
◇ Test testPtr() started.
◇ Test testRelativePath(_:) started.
◇ Test "caching items with the same keys but different namespaces" started.
◇ Test "cache removal and counting" started.
◇ Test testInvalidLoadingBundleFromDirectory(tc:) started.
◇ Test returnsUIntForPositiveWholeNumbers() started.
◇ Test testOverlap(tc:) started.
◇ Test testDataTreeContained(tc:) started.
◇ Test "rand.intn returns correct value for n=" started.
◇ Test testBuiltins(tc:) started.
◇ Test randIntNReturnsDifferentValueForDifferentBounds() started.
◇ Test testBuiltins(tc:) started.
◇ Test randIntNReturnsSameValueForSameInputs() started.
◇ Test randIntNReturnsDifferentValuesForDifferentKeys() started.
◇ Test testBuiltins(tc:) started.
◇ Test testBuiltins(tc:) started.
◇ Test testBuiltins(tc:) started.
◇ Test testBuiltins(tc:) started.
◇ Test timeNowNanosReturnsTimeFromContext() started.
◇ Test timeNowNanosReturnsDifferentValuesForDifferentContexts() started.
◇ Test timeNowNanosReturnsValidTime() started.
◇ Test timeNowNanosReturnsSameValueForSameContext() started.
◇ Test testBuiltins(tc:) started.
◇ Test testBuiltins(tc:) started.
◇ Test testBuiltins(tc:) started.
◇ Test testBuiltins(tc:) started.
◇ Test testBuiltins(tc:) started.
◇ Test testBuiltins(tc:) started.
◇ Test testBuiltins(tc:) started.
◇ Test rfc4122ReturnsSameValueForSameKey() started.
◇ Test rfc4122ReturnsValidUUID() started.
◇ Test testBuiltins(tc:) started.
◇ Test rfc4122ReturnsDifferentValuesForDifferentKeys() started.
◇ Test testBuiltins(tc:) started.
◇ Test testBuiltins(tc:) started.
◇ Test rfc4122ReturnsDifferentValuesForNewContext() started.
◇ Passing 1 argument tc → with metadata to testValidDecodeManifests(tc:)
◇ Passing 1 argument tc → more depdup to testResultSet(tc:)
◇ Passing 1 argument tc → simple to testValidDecodeManifests(tc:)
◇ Passing 1 argument tc → simple depdup to testResultSet(tc:)
◇ Passing 1 argument tc → rego v0 to testValidDecodeManifests(tc:)
◇ Passing 1 argument tc → rego v1 to testValidDecodeManifests(tc:)
◇ Passing 1 argument tc → empty roots to testValidDecodeManifests(tc:)
◇ Passing 1 argument tc → missing roots to testValidDecodeManifests(tc:)
◇ Passing 1 argument tc → nil roots to testValidDecodeManifests(tc:)
◇ Passing 1 argument tc → missing revision to testValidDecodeManifests(tc:)
◇ Passing 1 argument tc → ErrorCase(description: "all data no plan", sourceBundles: [file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/nested-data-trees], query: "", input: AST.RegoValue.object([:]), expectedError: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.noPlansFoundError)) to testInvalidEvaluations(tc:)
◇ Passing 1 argument tc → ErrorCase(description: "query not found in valid bundle", sourceBundles: [file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/simple-directory-bundle], query: "data.not.found.query", input: AST.RegoValue.object([:]), expectedError: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.unknownQuery)) to testInvalidEvaluations(tc:)
◇ Passing 1 argument tc → ErrorCase(description: "bundle with rego but no plan json", sourceBundles: [file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/simple-directory-no-plan-bundle], query: "", input: AST.RegoValue.object([:]), expectedError: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.noPlansFoundError)) to testInvalidEvaluations(tc:)
◇ Passing 1 argument tc → ErrorCase(description: "duplicate valid bundles", sourceBundles: [file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/simple-directory-bundle, file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/simple-directory-bundle], query: "data.app.rbac.allow", input: AST.RegoValue.object([:]), expectedError: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.bundleNameConflictError)) to testInvalidEvaluations(tc:)
◇ Passing 1 argument tc → ErrorCase(description: "bundle with invalid plan json", sourceBundles: [file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/invalid-plan-json-bundle], query: "", input: AST.RegoValue.object([:]), expectedError: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.bundleInitializationError)) to testInvalidEvaluations(tc:)
◇ Passing 1 argument tc → happy path basic policy with input allow to testValidEvaluations(tc:)
◇ Passing 1 argument tc → happy path basic policy with input deny explicit to testValidEvaluations(tc:)
◇ Passing 1 argument tc → happy path basic policy with input deny undefined input key to testValidEvaluations(tc:)
◇ Passing 1 argument tc → happy path rbac allow non-admin to testValidEvaluations(tc:)
◇ Passing 1 argument tc → happy path rbac allow admin to testValidEvaluations(tc:)
◇ Passing 1 argument tc → happy path rbac deny missing grant in data json to testValidEvaluations(tc:)
◇ Test testBuiltins(tc:) started.
◇ Passing 1 argument tc → invalid rego version to testInvalidDecodeManifests(tc:)
◇ Passing 1 argument tc → simple-directory-bundle to testDirectoryLoader(tc:)
◇ Passing 1 argument tc → simple-directory-bundle to testLoadingBundleFromDirectory(tc:)
◇ Passing 1 argument tc → nested-data-trees to testLoadingBundleFromDirectory(tc:)
◇ Passing 1 argument tc → BundleConsistencyTestCase(description: "simple bundle, roots allow all", sourceBundle: file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/simple-directory-bundle, rootsOverride: [""], isValid: true) to testBundleConsistency(tc:)
◇ Passing 1 argument tc → BundleConsistencyTestCase(description: "be explicit about the roots within data.json", sourceBundle: file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/simple-directory-bundle, rootsOverride: ["role_grants", "user_roles"], isValid: true) to testBundleConsistency(tc:)
◇ Passing 1 argument tc → BundleConsistencyTestCase(description: "be explicit about the roots one is out of bounds", sourceBundle: file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/simple-directory-bundle, rootsOverride: ["not_role_grants", "user_roles"], isValid: false) to testBundleConsistency(tc:)
◇ Passing 1 argument tc → BundleConsistencyTestCase(description: "roots internal conflict with wildcard", sourceBundle: file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/simple-directory-bundle, rootsOverride: ["", "user_roles"], isValid: false) to testBundleConsistency(tc:)
◇ Passing 1 argument tc → BundleConsistencyTestCase(description: "roots internal conflict", sourceBundle: file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/simple-directory-bundle, rootsOverride: ["role_grants", "user_roles", "user_roles/uh-oh"], isValid: false) to testBundleConsistency(tc:)
◇ Passing 1 argument tc → simple nested lookup to testStoreReads(tc:)
number(42)
◇ Passing 1 argument tc → key not found to testStoreReadsFailures(tc:)
◇ Passing 1 argument tc → Statement: assign local to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: assign local - source undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: assign local - overwrite to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: assign local - constant to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: assign local - string ref to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: initial assign local to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: assign local - source undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: reassign string ref - same value allowed to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: reassign string ref - different value is an error to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: break makes the block undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: breaking out of call frame is an error to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: infinite recursion to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: in-bounds array lookup to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: out-of-bounds array lookup is undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: negative out-of-bounds array lookup is undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: object lookup to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: object lookup - key not found to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: set lookup to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: set lookup - not found to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: unsupported type lookup - undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: len of empty array is 0 to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: len of non-empty array to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: len of empty set to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: len of non-empty set to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: len of empty string to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → TestCase(description: "simple relative with trailing slash", baseURL: file:///foo/bar/, childURL: file:///foo/bar/baz/qux, expected: "baz/qux") to testRelativePath(_:)
◇ Passing 1 argument tc → TestCase(description: "simple relative", baseURL: file:///foo/bar, childURL: file:///foo/bar/baz/qux, expected: "baz/qux") to testRelativePath(_:)
◇ Passing 1 argument tc → TestCase(description: "not file urls", baseURL: http://localhost:8080/foo/bar, childURL: http://localhost:8080/foo/bar/baz/qux, expected: "") to testRelativePath(_:)
◇ Passing 1 argument tc → Statement: len of non-empty string to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → TestCase(description: "not an descendent path", baseURL: file:///foo/bar, childURL: file:///nope/bar/baz/qux, expected: "") to testRelativePath(_:)
◇ Passing 1 argument tc → Statement: len of empty object to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → TestCase(description: "equal", baseURL: file:///foo/bar, childURL: file:///foo/bar, expected: ".") to testRelativePath(_:)
◇ Passing 1 argument tc → Statement: len of non-empty object to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: len of unsupported type - integer to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: len of unsupported type - null to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: len of undefined is undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: first insert succeeds to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: second insert with equal value succeeds to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: second insert with different values fails to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: target is not an object to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: key is undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: value is undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: source undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: scalar - undefined to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: empty array to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: array - some iterations were truthy to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: array - none were truthy to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: object - copy key/values into results to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: object - empty to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: set - copy values into result to testStatementEvaluation(tc:)
◇ Passing 1 argument tc → Statement: set - empty to testStatementEvaluation(tc:)
◇ Test testBuiltins(tc:) started.
◇ Passing 1 argument tc → ErrorCase(sourceBundle: file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/invalid-manifest-not-in-root, expectedError: Rego.BundleLoader.LoadError.unexpectedManifest(/)) to testInvalidLoadingBundleFromDirectory(tc:)
✔ Test undefinedAlwaysPropagates() passed after 0.027 seconds.
◇ Passing 1 argument tc → TestCase(paths: [["a", "b"]], otherPaths: [["c", "d"]], hasConflict: false) to testOverlap(tc:)
◇ Passing 1 argument tc → TestCase(paths: [["a", "b"]], otherPaths: [["a", "d"]], hasConflict: false) to testOverlap(tc:)
◇ Passing 1 argument tc → TestCase(paths: [["a", "b"]], otherPaths: [["a", "b", "c"]], hasConflict: true) to testOverlap(tc:)
◇ Passing 1 argument tc → TestCase(paths: [["a", "b", "c"]], otherPaths: [["a", "b"]], hasConflict: true) to testOverlap(tc:)
◇ Passing 1 argument tc → TestCase(paths: [["a", "b"], ["a", "c"]], otherPaths: [["a", "d"], ["a", "e"]], hasConflict: false) to testOverlap(tc:)
◇ Passing 1 argument tc → TestCase(paths: [["a", "b"], ["a", "c"]], otherPaths: [["a", "c"], ["a", "e"]], hasConflict: true) to testOverlap(tc:)
◇ Passing 1 argument tc → TestCase(paths: [["a", "b"], ["a", "c"]], otherPaths: [["a", "c", "z"], ["a", "e"]], hasConflict: true) to testOverlap(tc:)
◇ Passing 1 argument tc → TestCase(paths: [["a"]], otherPaths: [["a", "c", "z"]], hasConflict: true) to testOverlap(tc:)
◇ Passing 1 argument tc → TestCase(paths: [["a", "b"], ["a", "b", "c"]], otherPaths: [["a", "c", "z"]], hasConflict: false) to testOverlap(tc:)
◇ Passing 1 argument tc → ErrorCase(sourceBundle: file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/invalid-data-escaped-chroot, expectedError: Rego.BundleLoader.LoadError.dataEscapedRoot) to testInvalidLoadingBundleFromDirectory(tc:)
◇ Passing 1 argument tc → DataTreeContainedTestCase(paths: [["foo", "bar"]], data: AST.RegoValue.object([AST.RegoValue.string("foo"): AST.RegoValue.object([AST.RegoValue.string("bar"): AST.RegoValue.number(42)])]), isValid: true) to testDataTreeContained(tc:)
◇ Passing 1 argument tc → DataTreeContainedTestCase(paths: [[""]], data: AST.RegoValue.object([AST.RegoValue.string("foo"): AST.RegoValue.object([AST.RegoValue.string("bar"): AST.RegoValue.number(42)]), AST.RegoValue.string("baz"): AST.RegoValue.number(777)]), isValid: true) to testDataTreeContained(tc:)
◇ Passing 1 argument tc → DataTreeContainedTestCase(paths: [["foo", "bar"]], data: AST.RegoValue.object([AST.RegoValue.string("foo"): AST.RegoValue.object([AST.RegoValue.string("bar"): AST.RegoValue.number(42)]), AST.RegoValue.string("baz"): AST.RegoValue.number(777)]), isValid: false) to testDataTreeContained(tc:)
◇ Passing 1 argument tc → DataTreeContainedTestCase(paths: [["foo", "bar"]], data: AST.RegoValue.object([AST.RegoValue.string("foo"): AST.RegoValue.object([AST.RegoValue.string("baz"): AST.RegoValue.number(42)])]), isValid: false) to testDataTreeContained(tc:)
◇ Passing 1 argument tc → DataTreeContainedTestCase(paths: [["foo", "bar"]], data: AST.RegoValue.object([:]), isValid: true) to testDataTreeContained(tc:)
◇ Passing 1 argument n → -2147483658 to "rand.intn returns correct value for n="
◇ Passing 1 argument n → -5647870 to "rand.intn returns correct value for n="
◇ Passing 1 argument tc → DataTreeContainedTestCase(paths: [[""]], data: AST.RegoValue.object([AST.RegoValue.string("foo"): AST.RegoValue.number(42)]), isValid: true) to testDataTreeContained(tc:)
◇ Passing 1 argument n → -500 to "rand.intn returns correct value for n="
◇ Passing 1 argument tc → DataTreeContainedTestCase(paths: [], data: AST.RegoValue.object([AST.RegoValue.string("foo"): AST.RegoValue.number(42)]), isValid: true) to testDataTreeContained(tc:)
◇ Passing 1 argument n → 0 to "rand.intn returns correct value for n="
◇ Passing 1 argument n → 10 to "rand.intn returns correct value for n="
◇ Passing 1 argument n → 2147483647 to "rand.intn returns correct value for n="
◇ Passing 1 argument n → 2147483657 to "rand.intn returns correct value for n="
◇ Passing 1 argument tc → rand.intn: with n=0 to testBuiltins(tc:)
◇ Passing 1 argument tc → rand.intn: with n is not an integer to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: with n is not an integer at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: with n is not an integer at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentCountMismatch), message: "builtin evaluation error: operand 1 must be integer number but got floating-point number", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: wrong number of arguments (too few) to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: wrong number of arguments (too few) at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: wrong number of arguments (too few) at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentCountMismatch), message: "argument count mismatch (got: 0, want: 2)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: wrong number of arguments (too many) with 3 arguments to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: wrong number of arguments (too many) with 3 arguments at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: wrong number of arguments (too many) with 3 arguments at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentCountMismatch), message: "argument count mismatch (got: 3, want: 2)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: str argument has incorrect type - number to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - number at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - number at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: str, got: number, want: string)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: str argument has incorrect type - set to testBuiltins(tc:)
Caught error: dataEscapedRoot
Caught error: unexpectedManifest(app/.manifest -- file:///workspace/.build/aarch64-unknown-linux-gnu/debug/swift-opa_RegoTests.resources/TestData/Bundles/invalid-manifest-not-in-root/)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - set at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - set at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: str, got: set, want: string)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: str argument has incorrect type - object to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - object at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
◇ Passing 1 argument tc → rand.intn: str argument has incorrect type - null to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - object at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: str, got: object, want: string)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: str argument has incorrect type - boolean to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - null at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - boolean at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - null at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: str, got: null, want: string)", cause: nil)") == (String(reflecting: error) → "()")
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - boolean at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: str, got: boolean, want: string)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: str argument has incorrect type - undefined to testBuiltins(tc:)
◇ Passing 1 argument tc → rand.intn: str argument has incorrect type - array to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - array at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - array at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: str, got: array, want: string)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: n argument has incorrect type - object to testBuiltins(tc:)
◇ Passing 1 argument tc → rand.intn: n argument has incorrect type - set to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - undefined at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: str argument has incorrect type - undefined at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: str, got: undefined, want: string)", cause: nil)") == (String(reflecting: error) → "()")
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - object at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
◇ Passing 1 argument tc → rand.intn: n argument has incorrect type - null to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - set at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - object at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: n, got: object, want: number)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: n argument has incorrect type - boolean to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - set at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: n, got: set, want: number)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: n argument has incorrect type - string to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - null at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - boolean at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - null at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: n, got: null, want: number)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → rand.intn: n argument has incorrect type - undefined to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - boolean at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: n, got: boolean, want: number)", cause: nil)") == (String(reflecting: error) → "()")
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - string at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
◇ Passing 1 argument tc → rand.intn: n argument has incorrect type - array to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - string at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: n, got: string, want: number)", cause: nil)") == (String(reflecting: error) → "()")
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - undefined at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - array at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - undefined at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: n, got: undefined, want: number)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → array.concat: wrong number of arguments (too few) to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → rand.intn: n argument has incorrect type - array at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: n, got: array, want: number)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → array.concat: wrong number of arguments (too many) with 3 arguments to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → array.concat: wrong number of arguments (too few) at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → array.concat: wrong number of arguments (too many) with 3 arguments at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → array.concat: wrong number of arguments (too few) at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentCountMismatch), message: "argument count mismatch (got: 0, want: 2)", cause: nil)") == (String(reflecting: error) → "()")
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → array.concat: wrong number of arguments (too many) with 3 arguments at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentCountMismatch), message: "argument count mismatch (got: 3, want: 2)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → array.concat: x argument has incorrect type - number to testBuiltins(tc:)
◇ Passing 1 argument tc → array.concat: x argument has incorrect type - set to testBuiltins(tc:)
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → array.concat: x argument has incorrect type - number at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → array.concat: x argument has incorrect type - set at BuiltinTests.swift:62:13: Expectation failed: (type(of: expectedError) → RegoError) == (type(of: error) → ())
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → array.concat: x argument has incorrect type - number at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: x, got: number, want: array)", cause: nil)") == (String(reflecting: error) → "()")
✘ Test testBuiltins(tc:) recorded an issue with 1 argument tc → array.concat: x argument has incorrect type - set at BuiltinTests.swift:63:13: Expectation failed: (String(reflecting: expectedError) → "Rego.RegoError(code: Rego.RegoError.Code(internalCode: Rego.RegoError.Code.InternalCode.argumentTypeMismatch), message: "argument type mismatch (arg: x, got: set, want: array)", cause: nil)") == (String(reflecting: error) → "()")
◇ Passing 1 argument tc → array.concat: x argument has incorrect type - object to testBuiltins(tc:)
◇ Passing 1 argument tc → array.concat: x argument has incorrect type - boolean to testBuiltins(tc:)
◇ Passing 1 argument tc → array.concat: x argument has incorrect type - null to testBuiltins(tc:)
...
...
... (many more errors)
...
...
◇ Passing 1 argument tc → units.parse_bytes: 5e6 == 5000000 to testBuiltins(tc:)
✔ Suite "BuiltinRegistry" passed after 0.101 seconds.
✔ Suite "ResultSetTests" passed after 0.101 seconds.
✘ Test testBuiltins(tc:) failed after 0.073 seconds with 100 issues.
✔ Suite "BundleDecodingTests" passed after 0.101 seconds.
✔ Suite "IREvaluatorTests" passed after 0.101 seconds.
✔ Suite "NSNumber.toNumberRegoValue Extension" passed after 0.101 seconds.
✔ Suite BundleLoaderTests passed after 0.101 seconds.
✔ Suite "BundleDirectoryLoaderTests" passed after 0.101 seconds.
✔ Suite StoreTests passed after 0.101 seconds.
✔ Suite PtrTests passed after 0.101 seconds.
✔ Suite RelativePathTests passed after 0.101 seconds.
✔ Suite "BuiltinsCacheTests" passed after 0.101 seconds.
✔ Suite PathTests passed after 0.101 seconds.
✘ Suite "BuiltinTests - Rand" failed after 0.101 seconds with 34 issues.
✘ Suite "BuiltinTests - Crypto" failed after 0.100 seconds with 114 issues.
✘ Suite "BuiltinTests - Numbers" failed after 0.101 seconds with 90 issues.
✘ Suite "BuiltinTests - Array" failed after 0.101 seconds with 102 issues.
✘ Suite "BuiltinTests - Objects" failed after 0.101 seconds with 30 issues.
✘ Suite "BuiltinTests - Types" failed after 0.101 seconds with 32 issues.
✘ Suite "BuiltinTests - Time" failed after 0.101 seconds with 2 issues.
✘ Suite "BuiltinTests - Arithmetic" failed after 0.101 seconds with 73 issues (including 3 known issues).
✘ Suite "BuiltinTests - Casts" failed after 0.100 seconds with 30 issues.
✔ Suite "BuiltinTests - Collections" passed after 0.101 seconds.
✘ Test testBuiltins(tc:) failed after 0.011 seconds with 60 issues.
✘ Suite "BuiltinTests - UUID" failed after 0.101 seconds with 34 issues.
✘ Suite "BuiltinTests - Encoding" failed after 0.101 seconds with 152 issues.
✘ Suite "BuiltinTests - Strings" failed after 0.101 seconds with 476 issues.
✘ Suite "BuiltinTests - Comparison" failed after 0.101 seconds with 24 issues.
✘ Suite "BuiltinTests - Sets" failed after 0.101 seconds with 32 issues.
✘ Suite "CustomBuiltinTests" failed after 0.101 seconds with 34 issues.
✘ Suite "BuiltinTests - Bits" failed after 0.101 seconds with 200 issues.
✘ Suite "BuiltinTests - Units" failed after 0.101 seconds with 60 issues.
✘ Suite "BuiltinTests - Aggregates" failed after 0.101 seconds with 100 issues.
✘ Suite "BuiltinTests" failed after 0.101 seconds with 1619 issues (including 3 known issues).
✔ Test testStatementEvaluation(tc:) passed after 0.116 seconds.
✔ Suite IRStatementTests passed after 0.117 seconds.
✘ Test run with 58 tests failed after 0.118 seconds with 1619 issues (including 3 known issues).
```

</details>

## Proposed Solution

 - Fix known syntax issues breaking testing on 6.0.3.
 - Fix the spot in `Tests/RegoTests/BuiltinTests/BuiltinTests.swift` where the error isn't propagated properly.

--- a/Tests/RegoTests/BuiltinTests/BuiltinTests.swift
+++ b/Tests/RegoTests/BuiltinTests/BuiltinTests.swift
@@ -55,7 +55,7 @@ struct BuiltinTests {
         case .success:
             #expect(successEquals(result, tc.expected))
         case .failure(let expectedError):
-            let error = try #require(throws: (any Error).self, "Expect an error to be thrown") {
+            let error = try requireThrows(throws: (any Error).self, "Expect an error to be thrown") {
                 try result.get()
             }
 
@@ -100,7 +100,7 @@ struct BuiltinTests {
         wantArgs: String? = nil,
         generateNumberOfArgsTest: Bool = false,
         numberAsInteger: Bool = false,
-        builtinRegistry: BuiltinRegistry = .defaultRegistry,
+        builtinRegistry: BuiltinRegistry = .defaultRegistry
     ) -> [BuiltinTests.TestCase] {
         let argValues: [String: RegoValue] = [
             "array": [1, 2, 3], "boolean": false, "null": .null, (numberAsInteger ? "number[integer]" : "number"): 123,
@@ -134,7 +134,7 @@ struct BuiltinTests {
                     expected: .failure(
                         BuiltinError.argumentTypeMismatch(arg: argName, got: testType, want: want)
                     ),
-                    builtinRegistry: builtinRegistry,
+                    builtinRegistry: builtinRegistry
                 )
             )
         }
@@ -151,7 +151,7 @@ struct BuiltinTests {
     static func generateNumberOfArgumentsFailureTests(
         builtinName: String,
         sampleArgs: [RegoValue],
-        builtinRegistry: BuiltinRegistry = .defaultRegistry,
+        builtinRegistry: BuiltinRegistry = .defaultRegistry
     ) -> [BuiltinTests.TestCase] {
         var tests: [BuiltinTests.TestCase] = []
         // Only generate "too few" test case when expected number of arguments is > 0
@@ -163,7 +163,7 @@ struct BuiltinTests {
                     args: [],
                     expected: .failure(
                         BuiltinError.argumentCountMismatch(got: 0, want: sampleArgs.count)),
-                    builtinRegistry: builtinRegistry,
+                    builtinRegistry: builtinRegistry
                 )
             )
         }
@@ -179,7 +179,7 @@ struct BuiltinTests {
                 expected: .failure(
                     BuiltinError.argumentCountMismatch(
                         got: tooManyArgs.count, want: sampleArgs.count)),
-                builtinRegistry: builtinRegistry,
+                builtinRegistry: builtinRegistry
             )
         )
 
@@ -200,5 +200,25 @@ extension Result {
         } catch {
             self = .failure(error)
         }
+    }
+}
+
+// Note(philip): This function is needed to emulate modern #require behavior
+// in older versions of Swift (<= 6.0.3), returning the caught error. It should
+// be replaced with normal usage of #require as soon as 6.0.3 support is dropped.
+private func requireThrows<E: Error & Sendable, R>(
+    throws errorType: E.Type,
+    _ comment: String = "",
+    operation: () throws -> R
+) throws -> E {
+    do {
+        _ = try operation()
+        #expect(Bool(false), "Expected \(errorType) to be thrown. \(comment)")
+        fatalError("This should never be reached")
+    } catch let error as E {
+        return error
+    } catch {
+        #expect(Bool(false), "Expected \(errorType) but got \(type(of: error)). \(comment)")
+        fatalError("This should never be reached")
     }
 }

--- a/Tests/RegoTests/CapabilityEvaluatorTests.swift
+++ b/Tests/RegoTests/CapabilityEvaluatorTests.swift
@@ -55,7 +55,7 @@ struct CapabilityEvaluatorTests {
                 ))
         )
 
-        let error = try await #require(throws: RegoError.self, "Missing builtin must raise error") {
+        let error = try await requireThrows(throws: RegoError.self, "Missing builtin must raise error") {
             _ = try await engine.prepareForEvaluation(query: "policy")
         }
         #expect(error.code == .capabilitiesMissingBuiltin)
@@ -77,7 +77,7 @@ struct CapabilityEvaluatorTests {
                 ))
         )
 
-        let error = try await #require(throws: RegoError.self, "Mismatch builtin signature must fail") {
+        let error = try await requireThrows(throws: RegoError.self, "Mismatch builtin signature must fail") {
             _ = try await engine.prepareForEvaluation(query: "policy")
         }
         #expect(error.code == .capabilitiesMissingBuiltin)
@@ -125,7 +125,7 @@ struct CapabilityEvaluatorTests {
                 "my.slugify": { _, _ in .number(1) }
             ]
         )
-        let error = try await #require(throws: RegoError.self, "Missing builtin must raise error") {
+        let error = try await requireThrows(throws: RegoError.self, "Missing builtin must raise error") {
             _ = try await engine.prepareForEvaluation(
                 query: "policy"
             )
@@ -152,7 +152,7 @@ struct CapabilityEvaluatorTests {
             ]
         )
 
-        let error = try await #require(throws: RegoError.self, "Mismatched builtin signature must fail") {
+        let error = try await requireThrows(throws: RegoError.self, "Mismatched builtin signature must fail") {
             _ = try await engine.prepareForEvaluation(
                 query: "policy"
             )
@@ -176,7 +176,7 @@ struct CapabilityEvaluatorTests {
                 )),
             customBuiltins: [:]  // not specifying the builtin
         )
-        let error = try await #require(throws: RegoError.self, "Required builtin not provided must fail") {
+        let error = try await requireThrows(throws: RegoError.self, "Required builtin not provided must fail") {
             _ = try await engine.prepareForEvaluation(
                 query: "policy"
             )
@@ -198,7 +198,7 @@ struct CapabilityEvaluatorTests {
             capabilities: .path(IREvaluatorTests.relPath("TestData/Capabilities/does-not-exist.json"))
         )
 
-        let error = try await #require(throws: RegoError.self, "Unreadable capabilities file must raise error") {
+        let error = try await requireThrows(throws: RegoError.self, "Unreadable capabilities file must raise error") {
             _ = try await engine.prepareForEvaluation(query: "policy")
         }
         #expect(error.code == .capabilitiesReadError)
@@ -217,7 +217,7 @@ struct CapabilityEvaluatorTests {
             capabilities: .path(IREvaluatorTests.relPath("TestData/Capabilities/invalid.json"))
         )
 
-        let error = try await #require(throws: RegoError.self, "Invalid capabilities JSON must raise error") {
+        let error = try await requireThrows(throws: RegoError.self, "Invalid capabilities JSON must raise error") {
             _ = try await engine.prepareForEvaluation(query: "policy")
         }
         #expect(error.code == .capabilitiesDecodeError)
@@ -243,10 +243,30 @@ struct CapabilityEvaluatorTests {
             ]
         )
 
-        let error = try await #require(throws: RegoError.self, "Conflicting builtin name must raise error") {
+        let error = try await requireThrows(throws: RegoError.self, "Conflicting builtin name must raise error") {
             _ = try await engine.prepareForEvaluation(query: "policy")
         }
         #expect(error.code == .ambiguousBuiltinError)
         #expect(error.message.contains("count"))
+    }
+}
+
+// Note(philip): This function is needed to emulate modern #require behavior
+// in older versions of Swift (<= 6.0.3), returning the caught error. It should
+// be replaced with normal usage of #require as soon as 6.0.3 support is dropped.
+private func requireThrows<E: Error & Sendable, R>(
+    throws errorType: E.Type,
+    _ comment: String = "",
+    operation: () async throws -> R
+) async throws -> E {
+    do {
+        _ = try await operation()
+        #expect(Bool(false), "Expected \(errorType) to be thrown. \(comment)")
+        fatalError("This should never be reached")
+    } catch let error as E {
+        return error
+    } catch {
+        #expect(Bool(false), "Expected \(errorType) but got \(type(of: error)). \(comment)")
+        fatalError("This should never be reached")
     }
 }

